### PR TITLE
Prolog

### PR DESCRIPTION
**Pull request by itkovian**
Taken from the torque prologue.

### CODE REVIEW

**Comment by itkovian** [comment]
node2801

~~~~
[root@node2801 ~]# ls -l /var/tmp/
total 16
-rwx------ 1 root     root    21 Aug  8 15:27 checkpaths.cache.ts
drwx------ 3 vsc40075 root  4096 Aug  8 15:27 slurm.67119067.0
-rw-r--r-- 1 root     root     0 Aug  6 12:46 sssd_is_running
drwx------ 3 ageorges users 4096 Aug  8 14:54 yum-ageorges-eRRi5H
drwx------ 3 rvherck  users 4096 Aug  7 12:45 yum-rvherck-kk4ViQ
[root@node2801 ~]# cat  /var/tmp/checkpaths.cache.ts

1533734874 vsc40075
~~~~

~~~~
[2018-08-08T15:27:54.400] _run_prolog: run job script took usec=823254
[2018-08-08T15:27:54.400] _run_prolog: prolog with lock for job 67119067 ran for 1 seconds
~~~~

File: files/checkpaths_stat.sh [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+# we don not expect that many files in the apps dir anyway, so ls is ok
+mystat INSTITUTE_LOCAL 1
+
+
+# Site local and other tests
+uid=$(id -u)
+if [ "$uid" -gt 60000 ]; then
+    # Only for non-system users
+    if [ "$VSC_INSTITUTE" == "gent" ]; then
+        if [ "$VSC_INSTITUTE_CLUSTER" != "muk" ]; then
```
Comment (stdweird) [inline files/checkpaths_stat.sh]: remove this extra test

File: files/functions.sh [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+}
+
+function log () {
+    logger -- "$@"
+    debug "$@"
+}
+
+function set_drain () {
+    local reason="Prolog failure at job ${SLURM_JOB_ID} on $(date +%s): $1"
+
```
Comment (stdweird) [inline files/functions.sh]: can you add a logger line here?

**Comment by stdweird** [comment]
@itkovian this is good to go?

**Comment by itkovian** [comment]
I'll test again.

**Comment by itkovian** [comment]
Seems to work properly on banette.

**Pull request merged**